packages/server/src/fhircast/routes.ts

```diff
@@ -140,40 +117,92 @@ protectedCommonRoutes.post(
+      res.status(202).json({
+        'hub.channel.endpoint': getWebSocketUrl(config.baseUrl, `/ws/fhircast/${subscriptionEndpoint}`),
+      });
+      getRedis()
+        .publish(
+          topic,
+          JSON.stringify({
+            'hub.mode': 'denied',
+            'hub.topic': topic,
+            'hub.events': req.body['hub.events'],
+            'hub.reason': 'Subscriber unsubscribed from topic',
+          })
+        )
+        .catch((err: Error) => {
+          getLogger().error(`Error when publishing to Redis channel for FHIRcast topic: ${normalizeErrorString(err)}`, {
+            topic,
+          });
+        });
+    // break;
+  }
+}
+
+async function handleContextChangeRequest(req: Request, res: Response): Promise<void> {
+  const { event } = req.body as FhircastMessagePayload;
+  let stringifiedBody = JSON.stringify(req.body);
+  // Check if this an open event
+  if (event['hub.event'].endsWith('-open')) {
+    // TODO: Support this as a param for event type: "versionable"?
+    if (event['hub.event'] === 'DiagnosticReport-open') {
       event['context.versionId'] = generateId();
       stringifiedBody = JSON.stringify(req.body);
-      // TODO: Make sure this is actually supposed to be stored / overwrite open context? (ambiguous from docs, see: https://build.fhir.org/ig/HL7/fhircast-docs/2-9-GetCurrentContext.html)
-      await getRedis().set(`medplum:fhircast:topic:${req.params.topic}:latest`, stringifiedBody);
     }
-    await getRedis().publish(req.params.topic as string, stringifiedBody);
-    res.status(201).json({ success: true, event: body });
-  })
-);
+    // TODO: we need to get topic from event and not route param since per spec, the topic shouldn't be the slug like we have it
+    await getRedis().set(`medplum:fhircast:topic:${event['hub.topic']}:latest`, stringifiedBody);
+  } else if (event['hub.event'].endsWith('-close')) {
+    // We always close the current context, even if the event is not for the original resource... There isn't any mention of checking to see it's the right resource, so it seems it may be assumed to be always valid to do any arbitrary close as long as there is an existing context...
+    await getRedis().del(`medplum:fhircast:topic:${event['hub.topic']}:latest`);
+  } else if (event['hub.event'] === 'DiagnosticReport-update') {
+    // See: https://build.fhir.org/ig/HL7/fhircast-docs/3-6-3-DiagnosticReport-update.html#:~:text=The%20Hub%20SHALL,the%20new%20updates.
+    event['context.priorVersionId'] = event['context.versionId'];
+    event['context.versionId'] = generateId();
+    stringifiedBody = JSON.stringify(req.body);
+    // TODO: Make sure this is actually supposed to be stored / overwrite open context? (ambiguous from docs, see: https://build.fhir.org/ig/HL7/fhircast-docs/2-9-GetCurrentContext.html)
+    await getRedis().set(`medplum:fhircast:topic:${event['hub.topic']}:latest`, stringifiedBody);
+  }
+  await getRedis().publish(event['hub.topic'], stringifiedBody);
```

Comment:
The main change here is using `event['hub.topic']` instead of the `req.params.topic`